### PR TITLE
[UTILMAN] Implement the GetProcessID() helper

### PR DESCRIPTION
**Pull request by Fraizeraust**

### CODE REVIEW

File: base/applications/utilman/process.c [hunk 1]
```diff
@@ -44,17 +43,56 @@ BOOL IsProcessRunning(IN LPCWSTR lpProcessName)
...
+
+    /* Get the process ID */
+    dwProcessID = GetProcessID(lpProcessName);
+    if (dwProcessID == 0)
+    {
+        return FALSE;
+    }
+
+    /* Query some information from the process in question and check the exit code given by the process */
+    hProcess = OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, dwProcessID);
```
Comment (ThFabba) [inline base/applications/utilman/process.c]: Needs an error check.

Comment (ThFabba) [inline base/applications/utilman/process.c]: Does this actually succeed for zombie processes, or does the open already fail?

File: base/applications/utilman/process.c [hunk 2]
```diff
@@ -44,17 +43,56 @@ BOOL IsProcessRunning(IN LPCWSTR lpProcessName)
...
+    GetExitCodeProcess(hProcess, &dwExitCode);
```
Comment (ThFabba) [inline base/applications/utilman/process.c]: `GetExitCodeProcess` can fail (i.e. needs a return value check); but also, it's not the best way to check whether a process is running (what if a process exits with return code `STILL_ACTIVE`? See the documentation).
`WaitForSingleObject` with a zero timeout is a better indication (assuming you in fact get a valid handle, per the comment above).

File: base/applications/utilman/process.c [hunk 3]
```diff
@@ -147,43 +185,28 @@ BOOL LaunchProcess(LPCWSTR lpProcessName)
...
+    }
 
-    Process.dwSize = sizeof(Process);
+    /* Open the process so that we can terminate it */
+    hProcess = OpenProcess(PROCESS_TERMINATE, 0, dwProcessID);
 
-    /* Enumerate the processes */
-    if (Process32FirstW(hSnapshot, &Process))
+    /* Make sure that we do not terminate ourselves of course */
+    if ((hProcess != NULL) && (dwProcessID != GetCurrentProcessId()))
```
Comment (ThFabba) [inline base/applications/utilman/process.c]: Maybe add the `dwProcessID != GetCurrentProcessId()` check to the condition above? You'd still return FALSE, but avoid the unnecessary process open.

File: base/applications/utilman/process.c [hunk 4]
```diff
@@ -44,17 +43,62 @@ BOOL IsProcessRunning(IN LPCWSTR lpProcessName)
...
             {
-                /* The process we are searching for is running */
-                bIsRunning = TRUE;
-                break;
+                /* The names match, return the process ID we're interested */
+                CloseHandle(hSnapshot);
+                return Process.th32ProcessID;
             }
-        }
+        }
```
Comment (katahiromz) [inline base/applications/utilman/process.c]: Extra space

**Pull request merged**